packages/datetime/src/dateRangeInput.tsx

```diff
@@ -370,28 +410,45 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
     }
 
     private getSelectedRange = () => {
-        return [this.state.selectedStart, this.state.selectedEnd].map((selectedBound?: moment.Moment) => {
+        const momentDateRange = [this.state.selectedStart, this.state.selectedEnd] as MomentDateRange;
+        const [startDate, endDate] = momentDateRange.map((selectedBound?: moment.Moment) => {
             return this.isMomentValidAndInRange(selectedBound)
                 ? fromMomentToDate(selectedBound)
                 : undefined;
-        }) as DateRange;
+        });
+        // show only the start date if the dates overlap
+        // TODO: add different handling for the === case once
+        // allowSingleDayRange is implemented
+        return [startDate, (startDate >= endDate) ? null : endDate] as DateRange;
     }
 
-    private getStartInputDisplayString = () => {
-        return this.state.isStartInputFocused
-            ? this.state.startInputString
-            : this.getFormattedDateString(this.state.selectedStart);
-    }
+    private getInputDisplayString = (boundary: DateRangeBoundary) => {
+        const { values } = this.getStateKeysAndValuesForBoundary(boundary);
+        const { isInputFocused, inputString, selectedValue } = values;
 
-    private getEndInputDisplayString = () => {
-        return this.state.isEndInputFocused
-            ? this.state.endInputString
-            : this.getFormattedDateString(this.state.selectedEnd);
+        if (isInputFocused) {
+            return inputString;
+        } else {
+            if (isMomentNull(selectedValue)) {
```

Comment:
you can just move the whole if block outside of the else (i assume you're splitting it up the code like this to keep the logic distinction)